rakefile

```diff
@@ -612,4 +612,44 @@ namespace :pe_xl do
   task :deploy => [:clean, :provision, :run_plan] do
     puts "Deploying a pe_xl architecture"
   end
+
+end
+
+namespace :validate do
+  desc "Validate the environment variables"
+  task :validate_performance_env_vars do
+    Rake::Task["bigquery:verify_baseline_version"].invoke
+  end
+
+  desc "Verifies that the BEAKER_INSTALL_TYPE environment variable and dependencies have been specified"
+  task :verify_beaker_install_type do
+    # TODO: extract to a helper and spec test
+    # TODO: validate the specified values
+    raise "The BEAKER_INSTALL_TYPE env var must be set to either 'pe' or 'foss' to continue" unless
+        ENV['BEAKER_INSTALL_TYPE'] == 'pe' || ENV['BEAKER_INSTALL_TYPE'] == 'foss'
+
+    if ENV['BEAKER_INSTALL_TYPE'] == 'pe'
+      raise "The BEAKER_PE_VER and BEAKER_PE_DIR environment variables must be specified when BEAKER_INSTALL_TYPE == 'pe'" if
+          ENV['BEAKER_PE_VER'].nil? || ENV['BEAKER_PE_DIR'].nil?
+    else
+      raise "The PACKAGE_BUILD_VERSION and PUPPET_AGENT_VERSION environment variables must be specified when BEAKER_INSTALL_TYPE == 'foss'" if
+          ENV['PACKAGE_BUILD_VERSION'].nil? || ENV['PUPPET_AGENT_VERSION'].nil?
+    end
+  end
+
 end
+
+namespace :bigquery do
+  desc "List the baseline versions in BigQuery"
+  task :baseline_versions do
+    versions = baseline_versions
+    puts "Baseline versions:"
+    puts versions
+  end
+
+  desc "Verify that the specified baseline PE version exists in BigQuery"
+  task :verify_baseline_version do
+    # if BASELINE_PE_VER is specified verify that it exists before proceeding
+    verify_baseline_version ENV["BASELINE_PE_VER"] if ENV["BASELINE_PE_VER"]
```

Comment:
Seems like this shouldn't have an if...  cause if verify_baseline_version is called, it is implied that there is a version to verify.  The if should be on the thing that calls this task.  This may seem small, and it kinda is... but imagine a task that required a baseline_pe_ver.  It would have to do an if env var set check and this verify... where as if the if were not here, it could just call this verify and get a fail if unset or if not verified.  That makes the task more flexible and true to it's name.

Reply:
I've updated it to check BASELINE_PE_VER before calling the task.